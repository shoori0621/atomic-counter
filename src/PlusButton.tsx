import React from 'react';
import Button from './components/Button';

interface Props {
  count: number;
  setCounter: (count: number) => void;
}

export default function PlusButton(props: Props) {
  const {count, setCounter} = props;
  const addCounter = React.useCallback(() => {
    setCounter(count + 1);
  }, [count, setCounter]);
  return <Button label={'+'} onPress={addCounter} testId={'plus-btton'} />;
}
