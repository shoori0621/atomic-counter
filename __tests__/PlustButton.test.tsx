import React from 'react';
import renderer from 'react-test-renderer';
import {fireEvent, render, screen} from '@testing-library/react-native';
import PlusButton from '../src/PlusButton';

// Note: import explicitly to use the types shiped with jest.
import {it, describe, expect, jest} from '@jest/globals';

describe('PlusButton', () => {
  const count = 10;
  it('通常のPlusButtonを描画できる', () => {
    const component = renderer.create(
      <PlusButton count={count} setCounter={() => {}} />,
    );
    expect(component).toMatchSnapshot();
  });
  it(`ボタン押下でカウントが${count}から${count + 1}になる`, () => {
    const onPressEvent = jest.fn();
    render(<PlusButton count={count} setCounter={onPressEvent} />);
    fireEvent.press(screen.getByTestId('plus-button'));
    expect(onPressEvent).toBeCalled();
    expect(onPressEvent.mock.calls[0][0]).toBe(count + 1);
  });
});
