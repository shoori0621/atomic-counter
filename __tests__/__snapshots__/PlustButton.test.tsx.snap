// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlusButton 通常のPlusButtonを描画できる 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#008080",
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "width": 50,
    }
  }
  testID="plus-button"
>
  <Text
    style={
      {
        "color": "white",
        "fontSize": 20,
      }
    }
  >
    +
  </Text>
</View>
`;
